src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -316,6 +333,27 @@ private boolean isReadyToBeDeleted(Envelope envelope) {
         ).contains(envelope.getStatus());
     }
 
+    private void handleConstraintViolation(
+        String containerName,
+        String zipFilename,
+        String leaseId,
+        ConstraintViolationException exception
+    ) {
+        log.warn(
+            "Rejected file {} from container {} - DB constraint violation {}",
+            zipFilename,
+            containerName,
+            exception.getConstraintName(),
+            exception
+        );
+
+        if (exception.getConstraintName().equals("scannable_item_dcn")) {
+            handleInvalidFileError(Event.FILE_VALIDATION_FAILURE, containerName, zipFilename, leaseId, exception);
```

Comment:
I wonder - will the third party that uploaded the envelope will know why the file was invalid?

Reply:
the message will be in a form of

```java
new ErrorNotificationRequest(
            message.zipFileName,
            message.poBox,
            message.documentControlNumber,
            message.errorCode.name(),
            message.errorDescription,
            message.id
        );
```

where `errorDescription = exception.getMessage()` which from Constraint Exception transforms to `could not execute statement` :(. ok, perhaps need a better one and can re-use existing custom exception about duplicate dcns